src/accessibility/describe.js

```diff
@@ -0,0 +1,259 @@
+ * screen-reader users.
+ *
+ * @method describe
+ * @param  {String} text      description of the canvas
+ * @param  {Constant} [display] either LABEL or FALLBACK (Optional)
+ *
+ * @example
+ * <div>
+ * <code>
+ * describe('pink square with red heart in the bottom right corner', LABEL);
+ * background('pink');
+ * fill('red');
+ * noStroke();
+ * ellipse(66.6, 66.6, 20, 20);
+ * ellipse(83.2, 66.6, 20, 20);
+ * triangle(91.2, 72.6, 75, 95, 58.6, 72.6);
+ * </code>
+ * </div>
+ *
+ * <div>
+ * <code>
+ * let x = 0;
+ * function draw() {
+ *   if (x > 100) {
+ *     x = 0;
+ *   }
+ *   background(220);
+ *   fill(0, 255, 0);
+ *   ellipse(x, 50, 40, 40);
+ *   x = x + 0.1;
+ *   describe('green circle at x pos ' + round(x) + ' moving to the right');
+ * }
+ * </code>
+ * </div>
+ *
+ */
+p5.prototype.describe = function(t, d) {
+  p5._validateParameters('describe', arguments);
+  const cnvId = this.canvas.id;
+  //Creates a sub DOM inside of the canvas element and populates
+  //it with description text.
+  t = this._descriptionText(t);
+  if (document.getElementById(cnvId + '_Description') === null) {
+    document.getElementById(cnvId).innerHTML =
+      '<div id="' +
+      cnvId +
+      '_Description" role="region" aria-label="Canvas Description"><p id="' +
+      cnvId +
+      '_dsc"></p></div>';
+  } else if (document.getElementById(cnvId + '_dsc') === null) {
+    document
+      .getElementById(cnvId + '_eDsc')
+      .insertAdjacentHTML('beforebegin', '<p id="' + cnvId + '_dsc"></p>');
+  }
+  if (document.getElementById(cnvId + '_dsc').innerHTML !== t) {
+    document.getElementById(cnvId + '_dsc').innerHTML = t;
+  }
+  //If display is LABEL creates a div adjacent to the canvas element with
+  //description text.
+  if (d === 'label') {
+    if (document.getElementById(cnvId + '_Label') === null) {
+      document
+        .getElementById(cnvId)
+        .insertAdjacentHTML(
+          'afterend',
+          '<div id="' +
+            cnvId +
+            '_Label" class="p5Label"><p id=' +
+            cnvId +
+            '_dLbl></p></div>'
+        );
+    } else if (document.getElementById(cnvId + '_dLbl') === null) {
+      document
+        .getElementById(cnvId + '_eLbl')
+        .insertAdjacentHTML('beforebegin', '<p id=' + cnvId + '_dLbl></p>');
+    }
+    if (document.getElementById(cnvId + '_dLbl').innerHTML !== t) {
+      document.getElementById(cnvId + '_dLbl').innerHTML = t;
+    }
+  }
+};
+
+/**
+ * Helper function for describe() and describeElement().
+ */
+p5.prototype._descriptionText = function(t) {
+  if (t.endsWith('.') === false) {
+    t = t + '.';
+  }
+  if (/^[A-Z]/.test(t) === false) {
+    t = t[0].toUpperCase() + t.slice(1);
+  }
+  return t;
+};
+
+/**
+ * This function creates a screen-reader accessible
+ * description for elements —shapes or groups of shapes that create
+ * meaning together— in the canvas sub DOM. The first paramater should
```

Comment:
same comment here as above: suggest moving detail about the canvas sub DOM to the end of the description.